test/e2e/framework/autoscaling/autoscaling_utils.go

```diff
@@ -965,6 +966,11 @@ func CreateCustomResourceDefinition(ctx context.Context, c crdclientset.Interfac
 func ExistsInDiscovery(crd *apiextensionsv1.CustomResourceDefinition, apiExtensionsClient crdclientset.Interface, version string) (bool, error) {
 	groupResource, err := apiExtensionsClient.Discovery().ServerResourcesForGroupVersion(crd.Spec.Group + "/" + version)
 	if err != nil {
+		// Ignore 404 & 403 errors
+		// As they're not ignored by `ServerResourcesForGroupVersion` because passed version is not equal to `v1`
+		if version == "v1" && (errors.IsNotFound(err) || errors.IsForbidden(err)) {
```

Comment:
Wondering why `v1` is important here. Shouldn't we always ignore 404 and 403 in this test util? Even if the test creates a CRD with `v2`?

Reply:
For me, I didn't want to introduce any non-expected new behaviour, `v1` is already expected to ignore these errors but not for other versions

Reply:
I don't think it matters what `ServerResourcesForGroupVersion` does internally, their API has some inconsistencies due to backwards compatibility, but essentially if we've got 404 then the CRD doesn't exist in Discovery so we should return `false, nil`. Not sure about 403 though, does Discovery respond with 403 when it cannot find a CRD? Which HTTP code we get in flake runs?

Reply:
Flakiness so far is only caused by 404
However I'm not sure of 403 behaviour as I didn't see it happen

Well, I do agree with that actually :'D

-- Current edits to be made:
 1- Remove version check
 2- Only ignore when 404 errors occur
 
 Do you think there's anything else that should be made?

Reply:
I think that's gonna be perfect 👌

Reply:
Thanks 🙌 

- Keeping the convo unresolved till the new commit is reviewed